demos/background_subtraction_demo/cpp_gapi/main.cpp

```diff
@@ -82,6 +84,41 @@ static cv::gapi::GKernelPackage getKernelPackage(const std::string& type) {
     GAPI_Assert(false && "Unreachable code!");
 }
 
+cv::gapi::wip::onevpl::CfgParam createFromString(const std::string &line) {
+    using namespace cv::gapi::wip;
+
+    if (line.empty()) {
+        throw std::runtime_error("Cannot parse CfgParam from emply line");
+    }
+
+    std::string::size_type name_endline_pos = line.find(':');
+    if (name_endline_pos == std::string::npos) {
+        throw std::runtime_error("Cannot parse CfgParam from: " + line +
+                                 "\nExpected separator \":\"");
+    }
+
+    std::string name = line.substr(0, name_endline_pos);
+    std::string value = line.substr(name_endline_pos + 1);
+
+    return cv::gapi::wip::onevpl::CfgParam::create(name, value,
+                                                   /* vpp params strongly optional */
+                                                   name.find("vpp.") == std::string::npos);
+}
+
+static std::vector<cv::gapi::wip::onevpl::CfgParam> parseVPLParams(const std::string& cfg_params) {
+    std::vector<cv::gapi::wip::onevpl::CfgParam> source_cfgs;
+    std::stringstream params_list(cfg_params);
+    std::string line;
+    while (std::getline(params_list, line, ';')) {
+        source_cfgs.push_back(createFromString(line));
+    }
+    return source_cfgs;
+}
+
+static bool checkIfVPLSupportExt(const std::string& ext) {
+    return ext == "mp4" || ext == "avi";
```

Comment:
It might be some raw formats for `linux` such as `.h264`, `h265`... Probably it's better to remove this check and rely on `VPL` source error, instead of changing demo every time

Reply:
You can add to `-use_onevpl` description, that most likely input file must be .mp4 or .avi

Reply:
VPL source is immune for file extension:
for Windows it initially make check for container extension in file name and tries to make appropriate demultiplexing;
If demultiplexing should fail then it tries another types of demux for different extensions.
So, you may get actual MP4 container with extension related to *..avi file and VPLSource can determine the right one

BUT,  if you feed it with non-containerized video but  RAW video format: like pure h264, h265 and so on then file extension is not matter in this case, because it's raw video file. In this case you must specify code type in VPL args for your VPL Sorce in CfgParams.
For example, if you have RAW video with h265 then you may put in in file 'myVideo.any_extension' and just create CfgParams "**mfxImplDescription.mfxDecoderDescription.decoder.CodecID:MFX_HEVC_CODEC**" and pass it into source

Please pay attention, that containerized video like mp4, avi and etc mustn't require  any "mfxImplDescription.mfxDecoderDescription.decoder.CodecID" params. If you passed it with containerized video then you would get error, because VPL source would consider it as raw

Reply:
Removed